### ISSUE TITLE
Cannot read property 'path' of undefined

### ISSUE THREAD

**Issue by abhiyankhanal** [HashLips/hashlips_art_engine#124]
(node:5588) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'path' of undefined
    at D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:145:61
    at new Promise (<anonymous>)
    at loadLayerImg (D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:144:10)
    at D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:282:31
    at Array.forEach (<anonymous>)
    at startCreating (D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:281:17)
    at D:\fiverr\nft tool\hashlips_art_engine-main\index.js:13:3
    at Object.<anonymous> (D:\fiverr\nft tool\hashlips_art_engine-main\index.js:14:3)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:5588) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:5588) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:5588) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'path' of undefined
    at D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:145:61
    at new Promise (<anonymous>)
    at loadLayerImg (D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:144:10)
    at D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:282:31
    at Array.forEach (<anonymous>)
    at startCreating (D:\fiverr\nft tool\hashlips_art_engine-main\src\main.js:281:17)
    at D:\fiverr\nft tool\hashlips_art_engine-main\index.js:13:3
    at Object.<anonymous> (D:\fiverr\nft tool\hashlips_art_engine-main\index.js:14:3)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
(node:5588) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)

**Comment by tdev00** [comment]
I got the same issue. Even after using v1.7. What is the cause?

**Comment by nshCore** [comment]
I just came across this issue, 

you probably have duplicate files named in a similar fashion

green-layerx#10.png
green-layery#10.png
green-layerz#10.png

Its not documented in the readme but this naming convention of pre-appening a word with 

```
-
```

is used as the DNA_DELIMITER value which cant be duplicated, if you didn't intend to signify a dns trait just rename your files from 

```
-
```

with

```
_
```

**Comment by corbybender** [comment]
I ran into this issue this morning and after reading the above suggestions, I found that only 1 image in all of my layer files had a hyphen in the hame.  Any time my generator would hit that particular file, it would crash.  I found the image searching each layer with *-*.*  Once found, I renamed the file so there was no hyphen and now it generates hundreds with no failures.
Old File Name that caused the problem:  "Duck-feet#20.png"
New FIle Name that now works:  "Duckfeet#20.png"
